### ISSUE TITLE
`aria-hidden-focus` rule it is not showing an error

### ISSUE THREAD

**Issue by Luccasoli** [NickColley/jest-axe#279]
The tested code:

```jsx
 it.only('Should not have any accessibility violations', async () => {
    (useIsMobile as jest.Mock).mockReturnValue(false);
    expect.extend(toHaveNoViolations);

    const { container } = render(
      <div aria-hidden='true'>
        <input aria-disabled='true' />
      </div>
    );
    const results = await axe(container, {
      rules: {
        'aria-hidden-focus': { enabled: true },
      },
    });

    expect(results).toHaveNoViolations();
  });
```

![image](https://github.com/NickColley/jest-axe/assets/31039396/772cd789-ae42-4371-ba42-79ad5f28f434)

libs versions:
```json
{
"@types/jest-axe": "^3.5.9",
"jest-axe": "^6.0.1",
"@testing-library/dom": "^9.3.1",
"@testing-library/jest-dom": "^5.16.4",
"@testing-library/react": "^12.1.4",
}
```

**Comment by NickColley** [comment]
Please raise axe-core related issues with axe-core, if you find it's an issue with jest-axe itself then feel free to re-open later :)

https://github.com/dequelabs/axe-core
